zqd/packet/packet.go

```diff
@@ -183,7 +189,16 @@ func (p *IngestProcess) Done() <-chan struct{} {
 	return p.done
 }
 
-func (p *IngestProcess) writeData(ctx context.Context) error {
+type recWriter struct {
+	r *zng.Record
+}
+
+func (rw *recWriter) Write(r *zng.Record) error {
+	rw.r = r
+	return nil
+}
+
+func (p *IngestProcess) writeData(ctx context.Context) (nano.Ts, nano.Ts, error) {
```

Comment:
My preference would be generate the min/max cache when the `/space/:space/info` endpoint is hit and no space cache exists- its really the only place that cares about this information. That said I can see an argument for your approach, my approach would make the first call to space/info slow.

Reply:
Yes, this is discussed in the PR description.